cmake/sip_helper.cmake

```diff
@@ -200,10 +200,12 @@ function(build_sip_binding PROJECT_NAME SIP_FILE)
     set(SIP_BUILD_DIR ${sip_BINARY_DIR}/sip/${PROJECT_NAME})
 
     set(INCLUDE_DIRS ${${PROJECT_NAME}_INCLUDE_DIRS} ${PYTHON_INCLUDE_DIRS})
-    set(LIBRARIES ${${PROJECT_NAME}_LIBRARIES})
     set(LIBRARY_DIRS ${${PROJECT_NAME}_LIBRARY_DIRS})
     set(LDFLAGS_OTHER ${${PROJECT_NAME}_LDFLAGS_OTHER})
 
+    # SIP configure doesn't handle build configuration keywords
+    catkin_filter_libraries_for_build_configuration(LIBRARIES ${${PROJECT_NAME}_LIBRARIES})
```

Comment:
The package needs to declare a `buildtool_export` dependency on `catkin` in the manifest as well as export the dependency in CMake in order for this function to be able to rely on that function to be available.

Reply:
I followed this [guide](https://docs.ros.org/melodic/api/catkin/html/howto/format2/migrating_from_format_1.html#migrating-from-format1-to-format2) to upgrade the package format to v2 and added `buildtool_export_depend` on `catkin`. Let me know if I am doing it correct or wrong.